### DISCUSSION TITLE
CPACS 2 and 3 wing cutout shape is curved in TIGLviewer

### DISCUSSION THREAD

**Discussion by GodertdeZoetenGKNFokker** [DLR-SL/CPACS#741] [Q&A] [answered]
When I visualise the wing cutout of the innerFlap of the AGILE DC2 cpacs file (attached), I notice that the cutout of the upper surface is strangely curved. Attached you find a picture of this cutout (yellow) along with the flap upper surface (grey) which just connects the innerBorder and outerBorder with a straight line. Is there a reason why this is curved, and does the TIGL3viewer also do this?
[curved_cutout_issue.zip](https://github.com/DLR-SL/CPACS/files/5438089/curved_cutout_issue.zip)

**Comment by GodertdeZoetenGKNFokker** [comment]
Edit: This same behaviour is also seen with CPACS3 files:
[cpacs3_curved_cutout.zip](https://github.com/DLR-SL/CPACS/files/5438685/cpacs3_curved_cutout.zip)
Attached the CPACS3 file, along with two pictures of the cutout created by TIGL3viewer.
In these pictures it can be seen that the beginning and end of the cutout line is the same as the TED surface (grey), but it seems that some sort of BSpline curve is fitted through this begin and end point. Could it be that there are some other control points that influence this curve?

**Comment by MarAlder** [comment, marked as answer]
Hi Godert,

I think you did everything right. We know this this issue, however, for our preliminary aircraft design activities this has not been a big problem so far. I would say from the CPACS point of view the definition is sufficient. 

From a geometry point of view this rather raises the very complex question of how to cut the control surfaces with the skin of the wings. I know from a colleague of our Institute of Aerodynamics who is helping himself with a prototype using the TiGL Python-API, but I'm not aware of the status. 

For specific questions on the TiGL approach you might have a look at TiGL [GitHub](https://github.com/DLR-SC/tigl/issues). The developers know that there is room for improvement and welcome any prototype or suggestion.
